subsys/bluetooth/mesh/settings.c

```diff
@@ -116,6 +116,20 @@ struct mod_pub_val {
 	      cred:1;
 };
 
+#if defined(CONFIG_BT_MESH_PROVISIONER)
+/* Node storage information */
+struct node_val {
+	u16_t net_idx;
+	u8_t  dev_key[16];
+	u8_t  num_elem;
+} __packed;
+
+static struct node_update {
+	u16_t addr;
+	bool clear;
+} node_updates[CONFIG_BT_MESH_NODE_COUNT];
```

Comment:
We'd save a lot of memory by using two bitfields instead, one indicating change, and one indicating clear/store.

Reply:
How do you mean? Similar to the `struct key_update`? I'm not sure if a change flag is really needed for nodes?

Reply:
I meant changing the design from dynamically allocated list of pending changes to bitfields where each bit represents an index in the node array. We'd need one bit to represent "needs storing" and one to represent the type of change, as this can't be inferred from the `bt_mesh_node` alone, which might get reallocated before we're able to do the change. The settings entries must also be stored by index, instead of by address.

This'll reduce RAM requirements for each node by 15%, and reduce the amount of searching in this module. It'll also create a tighter coupling between the `bt_mesh` struct and the settings module though, so it's not all good :smile:

Reply:
Okay I see what you mean. I think it maybe could be another PR?

Reply:
Absolutely! :slightly_smiling_face:

Reply:
Yea that's true but I think it might be okay since this is a new addition and I think the API etc. will most likely change anyway. What do you think? @trond-snekvik